source/Cosmos.System2/FileSystem/Disk.cs

```diff
@@ -0,0 +1,152 @@
+﻿//#define COSMOSDEBUG
+
+using System;
+using System.Collections.Generic;
+using Cosmos.HAL.BlockDevice;
+using Cosmos.System.FileSystem.VFS;
+
+namespace Cosmos.System.FileSystem
+{
+    public class Disk
```

Comment:
Can we merge `PartitioningType` and `Disk` into the same class, since they seem to overlap a lot.

Reply:
We can, but the code will be messy if we do that.

Reply:
How so?